lib/new_relic/agent/llm/response_headers.rb

```diff
@@ -0,0 +1,28 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  module Agent
+    module Llm
+      module ResponseHeaders
+        ATTRIBUTES = %i[llm_version rate_limit_requests rate_limit_tokens
+          rate_limit_remaining_requests rate_limit_remaining_tokens
+          rate_limit_reset_requests rate_limit_reset_tokens]
+
+        attr_accessor(*ATTRIBUTES)
+
+        # Headers is a hash of Net::HTTP response headers
+        def populate_openai_response_headers(headers)
+          self.llm_version = headers['openai-version']&.first
+          self.rate_limit_requests = headers['x-ratelimit-limit-requests']&.first
+          self.rate_limit_tokens = headers['x-ratelimit-limit-tokens']&.first
+          self.rate_limit_remaining_requests = headers['x-ratelimit-remaining-requests']&.first
+          self.rate_limit_remaining_tokens = headers['x-ratelimit-remaining-tokens']&.first
+          self.rate_limit_reset_requests = headers['x-ratelimit-reset-requests']&.first
+          self.rate_limit_reset_tokens = headers['x-ratelimit-reset-tokens']&.first
```

Comment:
1. Without the `self.` prefix, a warning is printed to the console that the variable `llm_version` is defined but never used. This fails one of our unit tests. Alternatively, I could replace `self` with `@`.

2. Good point, I'll adjust

Reply:
For #1, sorry about that! Of course, without the `self` or `@` you'd be assigning a value to a local variable instead of lighting up the setter. Got it.

Reply:
Updated in: 2f40a7bf89cb36da1bbae664151960ab818f048d